### ISSUE TITLE
Make heatmaps more user friendly by supporting config params in heatmap.plot()

### ISSUE THREAD

**Issue by tanujkhattar** [quantumlib/Cirq#4017]
Currently, the heatmaps config parameters need to be passed during construction and can be updated by calling `heatmap.update_config`. This should be changed s.t. users can directly pass the parameters to `heatmap.plot()` function. Eg:
- Current workflow
```python
heatmap = calibration.heatmap("parallel_p11_error")
heatmap.update_config(vmin=0, vmax=0.1)
heatmap.plot()
```
- Expected Workflow
```python
heatmap = calibration.heatmap("parallel_p11_error")
heatmap.plot(vmin=0, vmax=0.1)
```

**Acceptance criteria - when is the task considered done?**
- Modify the heatmap code to support the updated workflow.
- Update visualization -> heatmap tutorial and add sections on how to use various config parameters. 

Part of the larger roadmap item: #3642

**Comment by rmlarose** [comment]
@tanujkhattar I think this should be closed now?

**Comment by tanujkhattar** [comment]
That's right, this should have been closed by #4026
